src/core/header.rs

```diff
@@ -0,0 +1,58 @@
+use crate::core::digest::Digests;
+use crate::core::{Hash, U256};
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, PartialEq, Eq)]
+pub struct Header<H: Hash, N: Copy + Into<U256> + TryFrom<U256>> {
+    /// the parent hash
+    pub parent_hash: H::Output,
+    /// block number or block height
+    pub number: N,
+    /// the root of state tired merkle tree
+    pub state_root: H::Output,
```

Comment:
What is this?

Reply:
This is the root of `Merkle Patricia Tree`. I prefer to use the `state model in ETH` to model the business in `zeeka` but it also bring some bad influence, for example: Expansion of data.